parlai/core/torch_ranker_agent.py

```diff
@@ -335,6 +357,8 @@ def is_valid(self, obs):
 
     def train_step(self, batch):
         """Train on a single batch of examples."""
+        if self.candidates != 'fixed' and self.opt.get('reencode_candidate_vecs'):
+            self.fixed_candidate_encs = None
```

Comment:
i'm not sure if this is doing anything here. if we pre-encode the candidates, they are must be passed in to the function `score_candidates(.., cand_encs=...)`. we are not passing them in during the train step (see line 376)

Reply:
we get `cand_encs` from `self.fixed_candidate_encs` during eval_step; i'm just deleting them here so that we know that we need to re-encode them during eval

Reply:
it seems a little weird to put this here, although i can't think of a better place. maybe just add a comment explaining?

@stephenroller  do you have any other ideas?

Reply:
Yes, pleaseeee add a comment. You could alternatively define a silly helper function:

```
def _invalidate_candidate_vec_cache(self):
    self.fixed_candidate_encs = None
```

so that the call is descriptive.